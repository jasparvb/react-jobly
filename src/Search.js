import React, { useState } from "react";

function Search({search}) {
    const INITIAL_STATE = { search: "" };
    const [formData, setFormData] = useState(INITIAL_STATE);
  
    /** Send {name, quantity} to parent
     *    & clear form. */
  
    const handleSubmit = evt => {
      evt.preventDefault();
      search({...formData});
    };
  
    /** Update local state w/curr state of input elem */
  
    const handleChange = evt => {
      const { name, value } = evt.target;
      setFormData(fData => ({
        ...fData,
        [name]: value
      }));
    };
  
    /** render form */
  
    return (
        <div className="Search mb-4">
            <form className="form-inline" onSubmit={handleSubmit}>
                <input className="form-control form-control-lg flex-grow-1" 
                    name="search" 
                    placeholder="Enter search term.." 
                    value={formData.search}
                    onChange={handleChange}
                />
                <button type="submit" className="btn btn-lg btn-primary">Submit</button>
            </form>
        </div>
    );
}

export default Search;