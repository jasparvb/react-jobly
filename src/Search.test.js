import React from "react";
import { render } from "@testing-library/react";
import Search from "./Search";

it("renders without crashing", function() {
  render(<Search />);
});

it("matches snapshot", function() {
  const { asFragment } = render(<Search />);
  expect(asFragment()).toMatchSnapshot();
});
