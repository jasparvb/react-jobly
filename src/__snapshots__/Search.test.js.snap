// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="Search mb-4"
  >
    <form
      class="form-inline"
    >
      <input
        class="form-control form-control-lg flex-grow-1"
        name="search"
        placeholder="Enter search term.."
        value=""
      />
      <button
        class="btn btn-lg btn-primary"
        type="submit"
      >
        Submit
      </button>
    </form>
  </div>
</DocumentFragment>
`;
